earth2mip/inference_medium_range.py

```diff
@@ -152,53 +150,28 @@ def process(initial_time):
 
             lead_time = valid_time - initial_time
             logger.debug(f"{valid_time}")
-            # TODO may need to fix n_history here
+            # TODO make this more performant grabs all history steps unnecessarily
             verification_torch = initial_conditions.get_initial_condition_for_model(
                 time_loop=model, data_source=data_source, time=valid_time
             )
             # select first history level
             verification_torch = verification_torch[:, 0]
-
-            output = {}
-            for name, metric in metrics.items():
-                output[name] = metric.call(verification_torch, data)
-            yield (initial_time, lead_time), output
-
-    # collect outputs for lead_times
-    my_channels = np.array(model.out_channel_names)
-    return metrics, my_channels, list(flat_map(process, initial_times))
-
-
-def gather(seq, metrics, model_name, channels):
-    outputs_by_lead_time = {}
-    initial_times = set()
-    for (initial_time, lead_time), metric_values in seq:
-        forecasts_at_lead_time = outputs_by_lead_time.setdefault(lead_time, [])
-        forecasts_at_lead_time.append(metric_values)
-        initial_times.add(initial_time)
-
-    def to_dataset(metric, name):
-        outputs = {
-            k: [v[name] for v in snapshots]
-            for k, snapshots in outputs_by_lead_time.items()
-        }
-        times, accs = zip(*outputs.items())
-        times = list(times)
-        acc_arr = [metric.gather(acc) for acc in accs]
-        stacked = torch.stack(acc_arr, 0)
-        stacked = stacked.cpu().numpy()
-        return xr.DataArray(
-            stacked,
-            dims=["lead_time", "channel"],
-            coords={"lead_time": times, "channel": channels},
-        ).to_dataset(name=name)
-
-    ds = xr.merge(to_dataset(metric, name) for name, metric in metrics.items())
-    ds = ds.assign(
-        initial_times=xr.DataArray(list(initial_times), dims=["initial_time"])
-    )
-
-    return ds
+            for metric in metrics:
+                outputs = metric.call(verification_torch, data)
+                for name, tensor in zip(metric.outputs, outputs):
```

Comment:
Nit: Interesting alternative to dictionary, I feel like `metric.outputs` should be called `metric.output_names` or keys or something

Reply:
done.